src/main/java/org/commcare/formplayer/configuration/WebSecurityConfig.java

```diff
@@ -49,6 +50,7 @@ protected void configure(HttpSecurity http) throws Exception {
         http.authorizeRequests().antMatchers("/**").authenticated();
         http.addFilterAt(getHmacAuthFilter(), UsernamePasswordAuthenticationFilter.class);
         http.addFilterAfter(sessionAuthFilter(), HmacAuthFilter.class);
+        http.csrf().csrfTokenRepository(CookieCsrfTokenRepository.withHttpOnlyFalse());
```

Comment:
I think you should update the filter to ignore requests that have the HMAC header (this will check if the header is present and the value is not `null`). These requests don't come from a browser, they come from HQ directly so CSRF isn't necessary.

```suggestion
        http.csrf()
            .csrfTokenRepository(CookieCsrfTokenRepository.withHttpOnlyFalse())
            .ignoringRequestMatchers(new RequestHeaderRequestMatcher(Constants.HMAC_HEADER));
```

We should also test to make sure that if the HMAC auth fails we don't fall back to the session auth otherwise that would allow you to provide any value for the HMAC header and bypass csrf.

It would be great if we could get an integration test for this. Some useful docs:
* https://www.baeldung.com/spring-security-integration-tests
* https://www.baeldung.com/spring-security-csrf